### ISSUE TITLE
Tokens being covered by object (door, furniture, painting ...) after save/load or undo

### ISSUE THREAD

**Issue by BenRQ** [NPBruce/valkyrie#891]
This bug is easily reproduced, when you have token on top of doors and furniture component.
Token should always been on top of objects as you need to click on them to be able to activate them.

Steps to reproduce : 
1) start a scenario with  token on top of doors and/or furniture component (example: A strain on reality)
2) do any action, then **undo** it
3) token are now under the furniture component, and cannot be selected (see capture)

No solution to fix it, you have to restart the game.

![screenshot_2018-09-10-09-50-16-248_com bruce valkyrie](https://user-images.githubusercontent.com/40794717/45320979-57c91680-b544-11e8-84ba-2153cbb68c1d.png)

### Valkyrie Version
Brynhildr 2.1.0 (but I'm sure it can be reproduced in 2.0)

**Comment by scrubbless** [comment]
I've seen this in 1.6.1 for sure.

**Comment by BenRQ** [comment]
So first of all, this issue happens when UNDOing, and when loading a quest.
This is basically the same action.

When loading a quest, the application is using the token order provided in the save file.
The first item loaded is on the bottom, the last one is on top.

For example : if a token of action is defined before a token of door, then the token of door will be on top.

**It means the order used by the scenario to define objects/token is quite important.**
The token that should be clickable should be defined last.

There is no easy way to identify token that should be on top from just other token in the code.
So I don't see any solution on app side yet.

@scrubbless in your case, you should make sure that the items with actions are defined after the doors and others items.

If someone has an idea for a fix,  please share !

**Comment by scrubbless** [comment]
The doors are placed in events then the tokens. But token order in the editor is listed based on when tokens were created.

That's a lot of reordering to do.

Could you bring all tokens that have events assigned (event=xxxx) to the top?

**Comment by BenRQ** [comment]
I would prefer to find a way to save the token items in the same order that they are displayed while playing.
I'm afraid reordering based on the event assigned may break something for someone else.

**Comment by scrubbless** [comment]
The problem is, how do you find this out? Tokens and Events are not sequencial and you can't always follow the event chain order to find a sequence.

Some event chains will place the same token too.

I guess you could just tag tokens every time they are placed with a number.

Then place them in lowest to highest order

**Comment by BenRQ** [comment]
After some time thinking and a bit of documentation reading, I found out the reason for this issue and the fix.

The reason : the type Dictionary<T1, T2> used to store the list of board items does not retain order.

The fix : The class OrderedDictionary does not exist unfortunately.
So I've added a 'sister' List of object name, and use this to write the savegame.

Tested and approved on "a strain for Reality".
